lib/array.rb

```diff
@@ -0,0 +1,15 @@
+class Array
+  def my_inject starting = nil, symbol = nil
+    starting.nil? ? result = self[0] : result = starting
+    if starting.is_a? Symbol
+      result = self[0]
+      each { |num| result = result.send starting, num }
+    elsif symbol.is_a? Symbol
+      each { |num| result = result.send symbol, num }
```

Comment:
These `if...else` statements are a great opportunity to split this method out into separate private methods. So we could do:

```ruby
return method_with_starting if starting.is_a? Symbol
...

def method_with_starting(starting)
  result = self[0]
  drop(0).each { |num| result = result.send starting, num }
  result
end
```